### ISSUE TITLE
What the best way to update CRS and reload spoa ?

### ISSUE THREAD

**Issue by mcarbonneaux** [jcmoraisjr/modsecurity-spoa#36]
there a way to update CRS and reload spoa without impact on haproxy using it ?

**Comment by mcarbonneaux** [comment]
i've synced to use git-sync and use the change hook to force reload, but how to force reload on spoa ?

**Comment by GMartinez-Sisti** [comment]
From what I know, all requests are sent to modsecurity and have a timeout to reply otherwise are allowed by default. This means you probably don’t need to reload SPOA as nothing changes in there.

**Comment by mcarbonneaux** [comment]
but the CRS are loaded at start of mod_sercurity sopa ?!

**Comment by GMartinez-Sisti** [comment]
Yes. Hot reload was added to modsecurity on 3.1.1 [here](https://github.com/SpiderLabs/ModSecurity/issues/648), this project uses [v2x](https://github.com/jcmoraisjr/modsecurity-spoa/blob/33570e2bd5dc10c36968e4475e6a9f58d0b4c370/rootfs/Dockerfile#L5) so you need to restart/kill the container to reload it.

I use this with a custom helm chart that adds checksum annotations for the config, so whenever the rules change all the modsecurity pods are automatically replaced by new ones on a deployment rollout. On this setup modsecurity is a Sidecar for haproxy so all the haproxies are also recycled along with it. If you configure pod disruption budgets and deployment rollout thresholds you shouldn’t have any downtime on update.

**Comment by GMartinez-Sisti** [comment]
If you manage to make modsecurity v3 working you can submit a PR. In theory that should work.
